gslib/commands/rsync.py

```diff
@@ -414,6 +414,11 @@
                  ignored. Note that gsutil does not follow directory symlinks,
                  regardless of whether -e is specified.
 
+  -i             This forces rsync to skip any files which exist on the destination
+                 and have a modified time that is newer than the source file.
+                 (If an existing destination file has a modification time equal to
+                 the source file's, it will be updated if the sizes are different.)
+
```

Comment:
As far as we can see in the implementation, modification time is not taken into consideration at all. But the description here states that rsync skips files only if the destination file “have a modified time that is newer than the source file,” which is misleading then.

If the author or any other project maintainer can confirm my concern, let us fix this.

P.S.
I've come here looking for an answer to the following question: [What is the difference between -i and -u options in gsutil rsync command?](https://serverfault.com/questions/1118444/what-is-the-difference-between-i-and-u-options-in-gsutil-rsync-command). And I've found out that we have this PR and earlier #583, which introduces the `-u` option, which indeed makes rsync skip the files only “if the destination mtime is newer.”

Reply:
At first glance, looks like you're right--there's no `>` or `<` symbols for the times in this PR.

However, mtime-related values are still returned by the PR code, and the parent function they're returned to is complex. We're redesigning `rsync` for the new `gcloud storage` CLI soon, and we'll probably wait until the accompanying in-depth study of gsutil `rsync` to confirm the behavior.

Reply:
Oh, and big thank you for catching this!

Reply:
> We're redesigning `rsync` for the new `gcloud storage` CLI soon

Do you mean that `rsync` will be **added** to the `gcloud storage` CLI? Or what kind of redesign do you mean? Currently I don't see `rsync` in the list of available collands: https://cloud.google.com/sdk/gcloud/reference/storage.

Reply:
Yes, we're still designing `rsync` for `gcloud storage`. It's taking a while because we're prototyping other algorithms to see if we can get performance improvements.